go/enclave/enclave.go

```diff
@@ -188,6 +188,11 @@ func NewEnclave(
 		genesis,
 		logger,
 	)
+	// ensure cached chain state data is up-to-date using the persisted batch data
```

Comment:
Should this happen on Start instead of on New ?

Reply:
Enclave doesn't have a start - agree that some of the stuff in this constructor probably belongs in a start though, but in a different PR perhaps?

Reply:
Fair point. Would add a TODO 👍